runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/DataUtil.java

```diff
@@ -119,8 +119,9 @@ private static void serializePartition(final EncoderFactory encoderFactory,
         // We need to close wrappedStream on here, because DirectByteArrayOutputStream:getBufDirectly() returns
         // inner buffer directly, which can be an unfinished(not flushed) buffer.
         wrappedStream.close();
+        // Note that serializedBytes include invalid bytes. So we have to use it with the actualLength whenever needed
```

Comment:
+1 for this comment. @hy00nc Could you add this comment to `getBufDirectly()` in `DirectByteArrayOutputStream`, too? Maybe we can also add todo like `// TODO: [NEMO-370] ...`

Reply:
I added the comment and TODO. Thanks for the review!